update-go-version.sh

```diff
@@ -14,5 +14,7 @@ go mod edit -go ${golang_base_version} tools/go.mod
 sed -i "s,^\(FROM registry.ci.openshift.org/openshift/release:rhel-8-release-golang-\)1.[0-9]\+,\1${golang_base_version}," images/*/Dockerfile
 sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registry.access.redhat.com/ubi8/go-toolset:${golang_base_version}," images/*/Dockerfile
 for f in .github/workflows/*.yml; do
-    yq eval --inplace ".jobs.build.strategy.matrix.go[0] = ${golang_base_version}" "$f";
+    if [ $(yq '.jobs.build.strategy.matrix | has("go")' "$f") == "true" ]; then
```

Comment:
but shouldn't we want to use latest always for yq?

Reply:
We have no check for the yq version, not even sure we have something to automatically install it, it's not packaged in fedora so won't be updated automatically...

In this case, the change is trivial and allows to support more `yq` versions, it will save us cryptic error messages if this ends up running somewhere with a slightly older yq version.

Reply:
> We have no check for the yq version, not even sure we have something to automatically install it, it's not packaged in fedora so won't be updated automatically...

It is packaged in fedora, got it from update repo
```
$ sudo dnf info yq
Available Packages
Name         : yq
Version      : 4.43.1
Release      : 2.fc39
Architecture : x86_64
Size         : 3.4 M
Source       : yq-4.43.1-2.fc39.src.rpm
Repository   : updates
Summary      : Yq is a portable command-line YAML, JSON, XML, CSV, TOML  and properties processor
URL          : https://github.com/mikefarah/yq
License      : MIT
Description  : Yq is a portable command-line YAML, JSON, XML, CSV, TOML  and properties
             : processor.
```
> 
> In this case, the change is trivial and allows to support more `yq` versions, it will save us cryptic error messages if this ends up running somewhere with a slightly older yq version.

Sure

Reply:
Done

Reply:
> It is packaged in fedora, got it from update repo

Looks like the first `yq` package landed in fedora updates yesterday ;) https://bugzilla.redhat.com/show_bug.cgi?id=2276522